### DISCUSSION TITLE
2FA not taking code

### DISCUSSION THREAD

**Discussion by 3lm0n0** [community/community#31679] [Other Feature Feedback, Questions, & Ideas] [answered]
I want to enable the 2 factor authentication but it's not taking the code.
![Screenshot_20220901-221859~2](https://user-images.githubusercontent.com/21112669/188038843-46760398-b93a-400a-bd91-314064460b6b.png)

**Comment by airtower-luna** [comment, marked as answer]
The code should be a 6 digit number generated by your TOTP application. That looks like you entered (and now shared!) the secret instead, so you should start over. The exact procedure will vary depending on what TOTP application you use, but at the core: Save the scanned code, and then tell it to generate a login code.

**Reply by 3lm0n0** [reply]
Thank you!

**Reply by airtower-luna** [reply]
You're welcome! And I'd appreciate if you mark the answer. :slightly_smiling_face:

**Reply by queenofcorgis** [reply]
Thanks @airtower-luna for the assistance and @3lm0n0 for marking the answer!

**Reply by ahmete60** [reply]
Doing that but not working

**Reply by 3lm0n0** [reply]
@ahmete60 could you share an error screenshot and explain in which step you are having trouble?

**Comment by GarthWilson** [comment]
Isn't the 16-character code shown there just the same thing that's in the QR code?  How're you supposed to turn that into a six-digit code?  What's the browser supposed to do with the 16-character one?  (And don't bother with "aps."  That's for smartphones, which I will not use.)

**Reply by airtower-luna** [reply]
The browser isn't supposed to do anything with it. You need some software that implements TOTP according to [RFC 6238](https://datatracker.ietf.org/doc/html/rfc6238), that code is the Base32 encoded secret to use for generating the one-time codes. [KeePassXC](https://keepassxc.org/) is a nice option if you don't want to build your own. Some hardware security tokens also support it.

**Reply by GarthWilson** [reply]
I can't find any such software in the Ubuntu Software Center that works.  I've installed several that don't require a smartphone or mobile device (which I won't use), and every one of them either failed to launch, or I got an error with no explanation, or you're supposed to click on something that doesn't respond.

**Reply by franklaranja** [reply]
Doesn't work for me using the secur app from fdroid. Get a 6 digit code but it fails at the github page.

**Comment by rao118417-hue** [comment]
Verify the code from the app

**Comment by LittleKenzy** [comment]
sigma

**Comment by LittleKenzy** [comment]
sigma

**Comment by zehraie1** [comment]
Hi there I have accidentially deleted 2fa authenticator but I have TOTP , I know my password bu but I can not log in my git hub account since asking 2fa code, I have created new account to log in this page, My actual account has been since 2022 and I need to reach out it 
I will appreciate If you support guidance please
